[GSoC 2025 - Acceptance Tests] M2.3 and M2.5 - Implement Acceptance tests for Translation Submitter/Reviewer and Question Submitter/Reviewer

jayam04 opened the pull request:
## Overview

<!--
READ ME FIRST:
Please answer *all* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes part of https://github.com/oppia/oppia/issues/22716.
2. This PR does the following:
    - Implements acceptance tests for Translation Submitter/Reviewer and Practice Question Submitter/Reviewer. ([CUJv3 Doc](https://docs.google.com/document/d/1D7kkFTzg3rxUe3QJ_iPlnxUzBFNElmRkmAWss00nFno/edit?tab=t.0))


## Essential Checklist

Please follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Make-a-pull-request).

- [ ] I have linked the issue that this PR fixes in the "Development" section of the sidebar.
- [ ] I have checked the "Files Changed" tab and confirmed that the changes are what I want to make.
- [ ] I have written tests for my code.
- [ ] The **PR title** starts with "Fix #bugnum: " or "Fix part of #bugnum: ...", followed by a short, clear summary of the changes.
- [ ] I have assigned the correct reviewers to this PR (or will leave a comment with the phrase "@{{reviewer_username}} PTAL" if I can't assign them directly).

## Testing doc (for PRs with Beam jobs that modify production server data)

<!--
If this PR affects production server data, please follow
[these instructions](https://github.com/oppia/oppia/wiki/Testing-jobs-and-other-features-on-production#submitting-a-pr-with-a-new-job-or-feature-that-requires-third-party-api)
and link to the job request doc here.

Otherwise, please delete this section.
-->

- [ ] A testing doc has been written: ... (ADD LINK) ...
- [ ] _(To be confirmed by the server admin)_ All jobs in this PR have been verified on a live server, and the PR is safe to merge.

## Proof that changes are correct

<!--
Add before-and-after videos/screenshots of the user-facing interface (including
the browser devtools console) to demonstrate that the changes made in this PR
work correctly. Make sure the actions taken in the before and after videos are
the same. (For changes involving responsiveness or adjustment of UI elements,
this should be a video that gradually resizes the viewport from wide to narrow
and back again.) If this PR is for a developer-facing feature, provide
videos/screenshots of the developer-facing interface instead.

When you make updates to the PR, please update these videos/screenshots as well.
You can drop videos/screenshots from previous versions of the PR.

The above should be done for all PRs, including short ones (e.g. a single-line change).
However, if the changes in your PRs are autogenerated via a script and you cannot
provide proof for the changes then please leave a comment "No proof of changes
needed because {{Reason}}" and remove all the sections below.
-->

#### Proof of changes on desktop with slow/throttled network

<!--
Make sure to properly verify that everything works correctly, and that there are
no weird UI mistakes or other problems. Also, if there are any newly added fields,
try to fill them out and test that different inputs are correctly accepted/rejected.

Throttle the network (to 3G) using the browser Developer Tools (see references below).
There should be no performance or UI issues while the network is slow.

References:
 - Chrome: https://css-tricks.com/throttling-the-network/
 - Firefox: https://developer.mozilla.org/en-US/docs/Tools/Network_Monitor/Throttling
-->

#### Proof of changes on mobile phone

<!--
In some cases this is not needed (e.g. for pages that we do not expect to
support mobile phones, or for backend-only features).

Feel free to use the Developer Tools emulator for this.

References:
 - Chrome: https://developer.chrome.com/docs/devtools/device-mode/
 - Firefox: https://firefox-source-docs.mozilla.org/devtools-user/index.html#responsive-design-mode
-->

#### Proof of changes in Arabic language

<!--
If the PR changes the UI, make sure to add screenshots with the site
language set to Arabic as well (we use Arabic as it is a language written from right to left).
-->

## PR Pointers

- Never force push! If you do, your PR will be closed.
- To reply to reviewers, follow these instructions: https://github.com/oppia/oppia/wiki/Make-a-pull-request#step-5-address-review-comments-until-all-reviewers-approve
- Some e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-CI-checks-fail-on-your-PR).
- See the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia's-code-owners-and-checks-to-be-carried-out-by-developers) for what code owners will expect.

jayam04 committed:
Implemented few steps.

jayam04 committed:
Updated full stack tests.yml

jayam04 committed:
removed other tests

jayam04 committed:
Updated full_stack_tests.yml.

jayam04 committed:
Fixes type errors.

jayam04 committed:
Added Steps (2)

jayam04 committed:
fixed subtopic not being created due to issue with name.

jayam04 committed:
Added Steps (3)

jayam04 committed:
Added Steps (4)

jayam04 committed:
Added Steps (5)

jayam04 committed:
increased timeout for beforeAll hook

jayam04 committed:
added few more steps, and fixed some parts for test

jayam04 committed:
added few more steps

jayam04 committed:
added few more steps

jayam04 committed:
updated full_stack_tests.yml

jayam04 committed:
added few more steps

jayam04 committed:
added some fixes, updated puppeteer-utils.ts

jayam04 committed:
updated tests to record video properly.

jayam04 committed:
Added few more steps

jayam04 committed:
Added test for translation reviewer.

jayam04 committed:
Added few more steps

jayam04 committed:
Updated exploration to use simple rte

jayam04 committed:
fix a step

jayam04 committed:
fix a step

jayam04 committed:
Added few more steps.

jayam04 committed:
Updated full_stack_tests.yml

jayam04 committed:
Added few fixed topic not being created.

jayam04 committed:
updated translate-exploration-in-target-language

jayam04 committed:
Updated function not to wait for welcome modal

jayam04 committed:
Added some fixes.

jayam04 committed:
Added a fix

jayam04 committed:
Added fix for other test.

jayam04 committed:
Added fixes

jayam04 committed:
Added few fixes

jayam04 committed:
Added fixes, new steps

jayam04 committed:
Updated clickOn function

jayam04 committed:
Added few steps

jayam04 committed:
Added few fixes

jayam04 committed:
Increased timeout

jayam04 committed:
Changed few steps

jayam04 committed:
Changed timeout from 10s to 1s for button check

jayam04 committed:
Added few fix

jayam04 committed:
Few fixes for translation submitter

jayam04 committed:
Few fixes for translation submitter  -- Part 2

jayam04 committed:
Added steps for translation reviewer

jayam04 committed:
translation submitter

jayam04 committed:
Fixes.

jayam04 committed:
Fixes

jayam04 committed:
Fixes

jayam04 committed:
Commented out most of the steps

jayam04 committed:
Fix

jayam04 committed:
Fix

jayam04 committed:
Fixes.

jayam04 committed:
Fixes

jayam04 committed:
Fixes.

jayam04 committed:
Logs

jayam04 committed:
Fixes

jayam04 committed:
Fixes

jayam04 committed:
Updated gitignore

jayam04 committed:
Fixes

jayam04 committed:
Updated full_stack_Tests.yml

jayam04 committed:
Updated tranlsation review

jayam04 committed:
New test and fixes.

jayam04 committed:
Fixes step in translation reviewer test

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/translation-reviewer/filter-the-translations.spec.ts:
```diff
@@ -0,0 +1,165 @@
+  }, 900000);
+
+  it('should be able to filter by topic', async function () {
+    // TODO: Can't see the one with different language.
```
Update the doc by adding suggestions and ping QA Team for reviews.

jayam04 committed:
Few fixes

jayam04 committed:
Added fix to failing step

jayam04 committed:
Fixes and screenshot.

jayam04 committed:
Increased timeout for before all in new test

jayam04 committed:
Merge remote-tracking branch 'upstream/develop' into m2.3

jayam04 committed:
Some Fixes

jayam04 committed:
Added screenshot check.

jayam04 committed:
Fixes

jayam04 committed:
Fixes CUJ 1

jayam04 committed:
Added step in 4th CUJ

jayam04 committed:
Updated full_stack_tests.yml

jayam04 committed:
Added a step

jayam04 committed:
Updates acceptance.json

jayam04 committed:
Added few more steps

jayam04 committed:
Added remaining steps.

jayam04 committed:
Debugging logs

jayam04 committed:
Added new test

jayam04 committed:
Updates acceptance.json

jayam04 committed:
Fixes

jayam04 committed:
Fixed a step in QS.CD.1

jayam04 committed:
Completed 2nd acceptance test

jayam04 committed:
Fixes

jayam04 committed:
Fixes.

jayam04 committed:
Fix

jayam04 committed:
Added one more test

jayam04 committed:
Added a comment

jayam04 committed:
Added last test

jayam04 committed:
Fixed error

jayam04 committed:
M2.3: Fixes upto step 2 of CUJ 1 of M2.3

jayam04 committed:
M2.3: Completly synced TS.CD.01

jayam04 committed:
M2.3: Completly synced TS.CD.02

jayam04 committed:
M2.3: Fixes

jayam04 committed:
Updated full_stack_tests.yml

jayam04 committed:
M2.3: Completly synced TR.CD.01

jayam04 committed:
Added edit state in translation review

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/practice-question-reviewer/review-submitted-questions.spec.ts:
```diff
@@ -0,0 +1,184 @@
+
+    await questionReviewer.startQuestionReview('What is 2 + 3?', 'Addition');
+    await questionReviewer.clickOn('Next');
+    await questionReviewer.expectQuestionInReviewModalToBe('12 + 14');
```
It should be "12 + 4"

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/practice-question-reviewer/review-submitted-questions.spec.ts:
```diff
@@ -0,0 +1,184 @@
+    await questionReviewer.expectQuestionInReviewModalToBe('12 + 14');
+  });
+
+  it('should be able to review the submitted quesions', async function () {
```
'questions'

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/practice-question-reviewer/review-submitted-questions.spec.ts:
```diff
@@ -0,0 +1,184 @@
+    // Modify and accept the question suggestion.
+    await questionReviewer.startQuestionReview('What is 2 + 3?', 'Addition');
+    await questionReviewer.editQuestionInReview('Updated Question');
+    // TODO: Update interaction type.
```
Missing

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/practice-question-reviewer/review-submitted-questions.spec.ts:
```diff
@@ -0,0 +1,184 @@
+      'Updated Question',
+      'Addition'
+    );
+    // TODO: View math interaction.
```
Missing

jayam04 replied:
Done.

jayam04 replied:
It should be Number Input (updated doc as well.)

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/practice-question-reviewer/review-submitted-questions.spec.ts:
```diff
@@ -0,0 +1,184 @@
+    // Accept the question suggestion.
+    await questionReviewer.startQuestionReview('What is 231 + 12?', 'Addition');
+    await questionReviewer.submitReview('accept', 'Test Review Message');
+    // TODO: Verify as topic manager.
```
Missing

jayam04 replied:
Done. It's same as above.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/practice-question-reviewer/review-submitted-questions.spec.ts:
```diff
@@ -0,0 +1,184 @@
+
+    // Accept the question suggestion.
+    await questionReviewer.startQuestionReview('What is 231 + 12?', 'Addition');
+    await questionReviewer.submitReview('accept', 'Test Review Message');
```
Missing the part of checking for "Updated Question" and "What is 231 + 12" is in the list. Also, checking if "What is 2 +3" is not exist in the list.

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/practice-question-reviewer/view-contribution-stats-and-badges-earned.spec.ts:
```diff
@@ -0,0 +1,192 @@
+
+  it('should be able to check badges earned', async function () {
+    await questionSubmitter.navigateToTabInMyContributions('Badges');
+    await questionSubmitter.expectBadgesToContain('1', 'Review');
```
Missing the part of `Expect 1 badge in the Corrections section with badge content “1 Correction”`

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/practice-question-submitter/check-contribution-stats-download-certificate-and-check-badges-earned.spec.ts:
```diff
@@ -0,0 +1,185 @@
+      'Addition',
+      'Arithmetic Operations',
+      'What is 2 + 3?'
+    );
```
The parts of `Add Multiple Choice Interaction with options “5” and “6”`. `Add response feedback to both responses`, and `Add Hint “1 + 2 = 3”` are missing.

jayam04 replied:
I have updated the doc. Also, adding hint and clicking on "Submit Question" is covered inside `startAndCompleteQuestionSuggestion`

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/practice-question-submitter/check-contribution-stats-download-certificate-and-check-badges-earned.spec.ts:
```diff
@@ -0,0 +1,185 @@
+    await questionSubmitter.startAndCompleteQuestionSuggestion(
+      'Addition',
+      'Arithmetic Operations',
+      '14 + 12'
```
Shouldn't it be `What is 14 + 12?` ?

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/practice-question-submitter/check-contribution-stats-download-certificate-and-check-badges-earned.spec.ts:
```diff
@@ -0,0 +1,185 @@
+    await questionSubmitter.switchToTabInContributionDashboard(
+      'My Contributions'
+    );
+    // Submit reviews.
```
These steps are not correct. This is the steps from the doc:
`1. Reject with review “No answer is correct”`
`2. Edit Title to “What is 14 + 12” and Accept`
`3. Accept without any edits.`

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/practice-question-submitter/check-contribution-stats-download-certificate-and-check-badges-earned.spec.ts:
```diff
@@ -0,0 +1,185 @@
+  });
+
+  it('should be able to download contribution certificate', async function () {
+    // TODO(ISSUE_NUMBER): Currently, the download certificate functionality is not working
```
Missing

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/practice-question-submitter/submit-practice-questions-for-review-and-check-thier-status.spec.ts:
```diff
@@ -0,0 +1,504 @@
+
+    // TODO: Contribution opportunities are not visible on the first page.
+    await questionSubmitter.expectScreenshotToMatch(
+      'emptyPracticeQuestionOpportunities',
```
Leave this comment so you can update this once all the screenshots are generated.

imchristie replied:
Still missing.

jayam04 replied:
Done.

jayam04 replied:
@imchristie This is complete right?

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/practice-question-submitter/submit-practice-questions-for-review-and-check-thier-status.spec.ts:
```diff
@@ -0,0 +1,504 @@
+      'Arithmetic Operations'
+    );
+    await questionSubmitter.expectScreenshotToMatch(
+      'practiceQuestionSubmissionTab',
```
Leave this comment so you can update this once all the screenshots are generated.

imchristie replied:
Still missing.

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/practice-question-submitter/submit-practice-questions-for-review-and-check-thier-status.spec.ts:
```diff
@@ -0,0 +1,504 @@
+      'Addition',
+      'Arithmetic Operations'
+    );
+    await questionSubmitter.selectQuestionDifficulty('Easy');
```
Based on the doc, it should be "medium" instead.

jayam04 replied:
Done.

jayam04 replied:
Sorry, I am reverting it. It seems test is better and we should update the doc.

jayam04 replied:
Done. Updated the doc.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/practice-question-submitter/submit-practice-questions-for-review-and-check-thier-status.spec.ts:
```diff
@@ -0,0 +1,504 @@
+      'Wrong Answer'
+    );
+    await questionSubmitter.addHintToState(
+      'If you have 2 apples and someone gives you 3 apples, how many apples do you have?'
```
Wrong hint compared to the doc

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/practice-question-submitter/submit-practice-questions-for-review-and-check-thier-status.spec.ts:
```diff
@@ -0,0 +1,504 @@
+  });
+
+  it('should be able to check question status', async function () {
+    // Accept the question suggestion.
```
Wrong steps. The steps in the doc are:
`1. Reject with review “No answer is correct”`
`2. Edit Title to “What is 14 + 12” and Accept`
`3. Accept without any edits.`

imchristie replied:
Either you follow the steps based on the doc or you update the doc.

jayam04 replied:
Done. Updated acceptance test.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/practice-question-submitter/submit-practice-questions-for-review-and-check-thier-status.spec.ts:
```diff
@@ -0,0 +1,504 @@
+    );
+    await questionSubmitter.submitQuestionSuggestion();
+
+    // Item Selection Interaction.
```
If "Item Selection" is the correct instead of "Image Selection", update the doc.

jayam04 replied:
Done. Updated the doc.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/practice-question-submitter/submit-practice-questions-for-review-and-check-thier-status.spec.ts:
```diff
@@ -0,0 +1,504 @@
+    );
+    await questionSubmitter.viewQuestionSuggestion('What is 10km + 11km?');
+    await questionSubmitter.expectScreenshotToMatch(
+      'questionContributionPreviewModal',
```
Leave this comment so you can update here once the screenshots are generated.

imchristie replied:
Still missing.

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/translation-reviewer/review-the-translations.spec.ts:
```diff
@@ -0,0 +1,213 @@
+  }, 900000);
+
+  it('should be able to view all pending reviews', async function () {
+    // TODO: Can't see the one with different language.
```
Why? Are you working on it?

jayam04 replied:
No. removed.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/translation-reviewer/review-the-translations.spec.ts:
```diff
@@ -0,0 +1,213 @@
+    await translationReviewer.navigateToContributorDashboardUsingProfileDropdown();
+    await translationReviewer.expectPinIconToBeVisible();
+    await translationReviewer.expectScreenshotToMatch(
+      'translationReviewerReviewTab',
```
Leave this comment so you can update once the screenshots are generated.

imchristie replied:
Still missing.

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/translation-reviewer/review-the-translations.spec.ts:
```diff
@@ -0,0 +1,213 @@
+    );
+    await translationReviewer.expectPaginationButtonToBeDisabled('previous');
+
+    await translationReviewer.clickOnPaginationButton('next');
```
The part of checking if `“Previous” button should be enabled.` is missing.

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/translation-reviewer/review-the-translations.spec.ts:
```diff
@@ -0,0 +1,213 @@
+
+    // Accept the translation with adding review comment.
+    await translationReviewer.clickOn('Edit');
+    // TODO(FILE_ISSUE): RTE not usable.
```
Did you file the issue?

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/translation-reviewer/review-the-translations.spec.ts:
```diff
@@ -0,0 +1,213 @@
+  it('should be able to accept the translation', async function () {
+    // Accept the translation without adding review comment.
+    await translationReviewer.submitTranslationReview('accept');
+
```
The part of checking if `Card content should change to the next card.` is missing

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/translation-reviewer/review-the-translations.spec.ts:
```diff
@@ -0,0 +1,213 @@
+      'accept',
+      'Review comment'
+    );
+
```
The part of checking if `Card content should change to the next card.` is missing

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/translation-reviewer/review-the-translations.spec.ts:
```diff
@@ -0,0 +1,213 @@
+    await translationReviewer.submitTranslationReview(
+      'accept',
+      'I have added some changes.'
+    );
```
The part of checking if `Card content should change to the next card.` is missing

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/translation-reviewer/review-the-translations.spec.ts:
```diff
@@ -0,0 +1,213 @@
+    await translationReviewer.submitTranslationReview(
+      'reject',
+      'Some lines are not translated properly'
+    );
```
The part of checking if `Review modals for the next card should appear.` is missing.

imchristie replied:
Still missing.

jayam04 replied:
As per test the review modal will be closed. I have updated the doc, as well as added a step below -- `expectReviewModalToBePresent`.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/translation-submitter/check-their-accomplishment.spec.ts:
```diff
@@ -0,0 +1,152 @@
+      'Translation Contributions'
+    );
+    await translationSubmitter.expectScreenshotToMatch(
+      'translationSubmitterAccomplishment',
```
Leave this comment so you can update once the screenshots are generated.

imchristie replied:
Still missing.

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/translation-submitter/check-their-accomplishment.spec.ts:
```diff
@@ -0,0 +1,152 @@
+  it('should be able to check badges earned', async function () {
+    await translationSubmitter.navigateToTabInMyContributions('Badges');
+    await translationSubmitter.expectScreenshotToMatch(
+      'translationSubmitterBadges',
```
Leave this comment so you can update once the screenshots are generated.

imchristie replied:
Still missing.

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/translation-submitter/translate-exploration-in-target-language.spec.ts:
```diff
@@ -0,0 +1,327 @@
+    await translationSubmitter.navigateToContributorDashboardUsingProfileDropdown();
+    await translationSubmitter.expectUsernameToBe('translator');
+    await translationSubmitter.expectScreenshotToMatch(
+      'contributorDashboard',
```
Leave this comment so you can update once the screenshots are generated.

imchristie replied:
Still missing.

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/translation-submitter/translate-exploration-in-target-language.spec.ts:
```diff
@@ -0,0 +1,327 @@
+    );
+    await translationSubmitter.expectTranslationOpportunitiesToBePresent(false);
+    await translationSubmitter.expectScreenshotToMatch(
+      'translationTabInContributionDashboard',
```
Leave this comment so you can update once the screenshots are generated.

imchristie replied:
Still missing.

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/translation-submitter/translate-exploration-in-target-language.spec.ts:
```diff
@@ -0,0 +1,327 @@
+    await translationSubmitter.expectPaginationButtonToBeVisible('next');
+    await translationSubmitter.expectPaginationButtonToBeVisible(
+      'previous',
+      false
```
Shouldn't it be true instead?

jayam04 replied:
On first page, the previous button won't be visible.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/translation-submitter/translate-exploration-in-target-language.spec.ts:
```diff
@@ -0,0 +1,327 @@
+    );
+  });
+
+  it('should be able to use RTE', async function () {
```
The part of checking if `The content of the RTE editor should change after using each of the components.` is missing.

jayam04 replied:
`typeTextForRTE` already has a post-check which checks for the changes.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/translation-submitter/translate-exploration-in-target-language.spec.ts:
```diff
@@ -0,0 +1,327 @@
+      'छवि विवरण',
+      'तस्वीर का शीर्षक'
+    );
+  });
```
The part of 
`Click on the “On” button next to the copy tool.`
`Select the image from the original content.`
`Add image description and caption in the pop-up modal.`
to check if 
`A new image should be added in the translated content box.`
is missing

jayam04 replied:
It's inside `expectCopyToolWorksProperly`. Do you mean something is missing in this function?

imchristie commented on core/tests/puppeteer-acceptance-tests/utilities/common/puppeteer-utils.ts:
```diff
@@ -147,12 +148,17 @@ export class BaseUser {
           const uniqueString = Math.random().toString(36).substring(2, 8);
           const outputFileName =
-            `${mobile ? 'mobile' : 'desktop'}-${specName}-${new Date().toISOString()}-${uniqueString}.mp4`.replace(
+            `${this.username}-${new Date().toISOString()}-${uniqueString}.mp4`.replace(
```
This will be converted back to `{mobile ? 'mobile' : 'desktop'}-${specName}`, right?

jayam04 replied:
No, I have updated folder name to be mobile-TESTSUITE and desktop-TESTSUITE.

imchristie commented on core/tests/puppeteer-acceptance-tests/utilities/common/rte-editor.ts:
```diff
@@ -164,4 +146,80 @@ export class RTEEditor {
+   * @param {string} content - The content to update the editor with.
+   */
+  async updateAndSaveContent(content: string): Promise<void> {
+    await this.parentPage.type(rteTextAreaSelector, content);
```
Missing pre-check

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/utilities/user/contributor.ts:
```diff
@@ -0,0 +1,593 @@
+    let opportunityItemListChanged = true;
+
+    do {
+      await this.page.waitForTimeout(200);
```
Hardcode the timeout is not a good practice, can you try other ways like waiting for a specific selector?

jayam04 replied:
Can you help me how can I achieve this? I am seeing that the elements refreshes after few mili-seconds. So, I created this function, it checks for chances every 200ms.

imchristie replied:
I need more context of what's going on on this page. Could you record a video? Does the element refresh once or multiple times? Why do you need to check for every 200ms?

jayam04 replied:
@imchristie I tried to look for it again, however, I didn't found the case. It got me thinking if the issue was due to filtering the content. I will look onto it and remove this if required.

jayam04 replied:
Done. Keeping it as you suggested in chat. Also added a todo comment above.

imchristie commented on core/tests/puppeteer-acceptance-tests/utilities/user/contributor.ts:
```diff
@@ -0,0 +1,593 @@
+  }
+
+  /**
+   * Expects the contribution table to contain a row with the given topic name,
```
Wrong parameters

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/utilities/user/contributor.ts:
```diff
@@ -0,0 +1,593 @@
+          el.querySelector(sel)?.textContent?.trim(),
+        badgeValueSelector
+      );
+      console.log(`[debug] badgeValue: ${badgeValue}`);
```
Make sure to remove all these [debug] console log before merging.

jayam04 replied:
Done. Yes, I will remove all. If there are any, lint checks won't pass.

imchristie replied:
@jayam04 Reminder

imchristie commented on core/tests/puppeteer-acceptance-tests/utilities/user/practice-question-reviewer.ts:
```diff
@@ -0,0 +1,132 @@
+// limitations under the License.
+
+/**
+ * @fileoverview Question Submitters utility file.
```
Practice Question Reviewer

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/utilities/user/practice-question-reviewer.ts:
```diff
@@ -0,0 +1,132 @@
+    await this.expectModalTitleToBe(skill);
+  }
+
+  async submitReview(
```
Missing description

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/utilities/user/practice-question-reviewer.ts:
```diff
@@ -0,0 +1,132 @@
+    reviewType: 'accept' | 'reject',
+    reviewMessage?: string
+  ): Promise<void> {
+    const buttonSelector = `.${reviewButtonPrefix}-${reviewType}-button`;
```
Missing pre-check

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/utilities/user/practice-question-reviewer.ts:
```diff
@@ -0,0 +1,132 @@
+    await this.expectToastMessage('Submitted suggestion review.');
+  }
+
+  async editQuestionInQuestionEditorModal(question: string): Promise<void> {
```
Missing description

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/utilities/user/practice-question-submitter.ts:
```diff
@@ -156,21 +170,18 @@ export class QuestionSubmitter extends BaseUser {
-      throw new Error(`Invalid difficulty level: ${difficulty}`);
-    }
+    const skillDifficultySelector = `.e2e-test-skill-difficulty-${difficulty.toLocaleLowerCase()}`;
+    // TODO(FILE_ISSUE): Currently, the difficulty selector is not visible.
```
Did you file the issue?

imchristie replied:
Still missing.

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/utilities/user/practice-question-submitter.ts:
```diff
@@ -484,6 +496,22 @@ export class QuestionSubmitter extends BaseUser {
     showMessage(`${interactionToAdd} interaction has been added successfully.`);
   }
 
+  async addResponseDetailsInQuestionResponseModal(
```
Missing description

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/utilities/user/practice-question-submitter.ts:
```diff
@@ -588,7 +616,75 @@ export class QuestionSubmitter extends BaseUser {
+    defaultResponseFeedback?: string,
+    hint?: string
+  ): Promise<void> {
+    await this.suggestQuestionsForSkillandTopic(skill, topic);
```
Missing pre-check

jayam04 replied:
`suggestQuestionsForSkillandTopic` has pre-check. Do I need to add another specifically for this function?

imchristie commented on core/tests/puppeteer-acceptance-tests/utilities/user/practice-question-submitter.ts:
```diff
@@ -588,7 +616,75 @@ export class QuestionSubmitter extends BaseUser {
+      hint ??
+        'If you have 2 apples and someone gives you 3 apples, how many apples do you have?'
+    );
+    await this.submitQuestionSuggestion();
```
Missing post-check

jayam04 replied:
`submitQuestionSuggestion` has it's post-check. Won't it serve as post-check for this function too?

imchristie replied:
Yes it counts as post-check. In the function `submitQuestionSuggestion`, could you add the `await` keyword for the last line?

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/utilities/user/translation-admin.ts:
```diff
@@ -68,7 +68,6 @@ export class TranslationAdmin extends BaseUser {
    * Function for adding a translation right to a user.
    */
-
   async addTranslationLanguageReviewRights(
```
Missing descriptions for the parameters.

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/utilities/user/translation-reviewer.ts:
```diff
@@ -0,0 +1,247 @@
+// limitations under the License.
+
+/**
+ * @fileoverview Utilty class for translation submitter.
```
translation reviewer

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/utilities/user/translation-reviewer.ts:
```diff
@@ -0,0 +1,247 @@
+    chapterName: string,
+    subheading: string
+  ): Promise<void> {
+    const opportunityItem = await this.getTranslationOpportunityCard(
```
Missing pre-check

jayam04 replied:
`getTranslationOpportunityCard` will check for element required before executing, so I think it will serve as pre-check.

imchristie commented on core/tests/puppeteer-acceptance-tests/utilities/user/translation-submitter.ts:
```diff
@@ -0,0 +1,516 @@
+
+    await this.clickOn(selector);
+
+    // TODO: Post-check: Verify if the page is loaded.
```
Leave this comment so you can update once it's ready.

imchristie replied:
Still missing.

jayam04 replied:
Done.

jayam04 replied:
@imchrisite don't resolve it. I am reverting the change.

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/utilities/user/translation-submitter.ts:
```diff
@@ -0,0 +1,516 @@
+   * @param title - The title of RTE option.
+   */
+  async clickOnRTEOptionContainingTitle(title: string) {
+    const rteEditor = new RTEEditor(this.page, this.page);
```
Missing pre-check

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/utilities/user/translation-submitter.ts:
```diff
@@ -0,0 +1,516 @@
+    await this.expectElementToBeVisible(selector, visible);
+  }
+
+  async expectCopyToolWorksProperly(
```
Missing description

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/utilities/user/translation-submitter.ts:
```diff
@@ -0,0 +1,516 @@
+    );
+    return;
+
+    // TODO: remove.
```
Leave this comment so you can update once it's completed.

jayam04 replied:
Done.

imchristie reviewed (commented):
Completed full review.

masterboy376 reviewed (commented):
Thanks @jayam04,
files owned by @oppia/contributor-experience-team looks good. However, I have few comments and questions:
1. PR description is not complete.  
2. I noticed couple of GitHub actions workflow files are deleted, could you provide the reason for that.

jayam04 committed:
Merge remote-tracking branch 'upstream/develop' into m2.3

jayam04 committed:
Addresses some quick comments

jayam04 commented:
> Thanks @jayam04, files owned by @oppia/contributor-experience-team looks good. However, I have few comments and questions:
> 
>     1. PR description is not complete.
> 
>     2. I noticed couple of GitHub actions workflow files are deleted, could you provide the reason for that.

@masterboy376 I have updated the PR description.
Yes, I will revert all workflow files. I have deleted them, so I can test my changes quickly.

jayam04 committed:
Addressed Comments (2) + Fixes some failing tests (1)

jayam04 committed:
Addressed few more comments

jayam04 committed:
Addresses few comments

jayam04 committed:
Some fixes

jayam04 committed:
Few uncommitted changes

jayam04 committed:
Addressed few more comments

jayam04 committed:
Merge remote-tracking branch 'upstream/develop' into m2.3

jayam04 committed:
Resolved UserFactory errors due to merge

jayam04 committed:
Addressed comments

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/logged-out-user/play-lesson-in-different-languages-and-listen-to-voiceovers.spec.ts:
```diff
@@ -49,162 +49,162 @@ ConsoleReporter.setConsoleErrorsToIgnore([
-  let explorationEditor: ExplorationEditor;
-  let curriculumAdmin: CurriculumAdmin;
-  let releaseCoordinator: ReleaseCoordinator;
+  // let explorationEditor: ExplorationEditor;
```
Why are all these commented out?

jayam04 replied:
Reverted.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/practice-question-submitter/submit-practice-questions-for-review-and-check-thier-status.spec.ts:
```diff
@@ -0,0 +1,492 @@
+    // Go to the contribution dashboard.
+    await questionSubmitter.navigateToContributorDashboardUsingProfileDropdown();
+
+    // TODO: Contribution opportunities are not visible on the first page.
```
Are you working on this?

jayam04 replied:
No removed.

imchristie reviewed (changes requested):
Completed full review

jayam04 committed:
Fixes

jayam04 committed:
Fixes: Mobile tests

jayam04 committed:
fixes

jayam04 committed:
Fixes some tests

jayam04 committed:
Fixes

jayam04 committed:
Fixes: First three (2)

jayam04 committed:
Updated full_stack_tests.yml

jayam04 committed:
Fixes: Last three (2)

jayam04 committed:
Fixes all Desktop

jayam04 committed:
Fixed a duplicate function

jayam04 committed:
Fixes

jayam04 committed:
Update puppeteer recording to not follow new page

jayam04 committed:
Fixes

jayam04 committed:
Fixes the test

jayam04 committed:
Fixes the test

jayam04 committed:
Fix: Test 7th

jayam04 committed:
Fixes desktop tests

jayam04 committed:
Mobile Fix: 6, 8

imchristie commented:
Friendly ping @kevintab95

jayam04 committed:
fixes

jayam04 committed:
Fixes

jayam04 committed:
fix post-check

jayam04 committed:
Desktop fixes

jayam04 committed:
Fixes

jayam04 committed:
Fixes

jayam04 committed:
Fix: addImageInteractionInQuestionEditor (mobile)

jayam04 committed:
Fix: selectContributionTypeInContributionDashboard

jayam04 committed:
Fix: addImageInteractionInQuestionEditor (desktop)

jayam04 committed:
Fix: addImageInteractionInQuestionEditor (mobile)

jayam04 committed:
Potential Fix: createSkillForTopic

jayam04 committed:
Fix: First translate in target language

jayam04 committed:
Fix: navigateToTabInMyContributions (mobile)

jayam04 committed:
Revert It: Updates full_stack_tests

jayam04 committed:
Fix: navigateToTabInMyContributions

jayam04 committed:
Fix: addTextInputInteractionInQuestionEditor (mobile)

jayam04 committed:
Fix: 2 mobile tests

jayam04 committed:
Fix: Mobile test

jayam04 committed:
Reverts full_stack_tests.yml

jayam04 committed:
Revert: 9e7f820ff8a386b802091b3fb89480d835a38ba8

jayam04 committed:
Few Efficiencies

jayam04 committed:
Fix

jayam04 committed:
Fix: Type 'Drag And Drop Sort'

jayam04 committed:
Fix: translate in target language

jayam04 committed:
Addresses few comments

jayam04 committed:
Addresses comment

jayam04 committed:
Reverts some steps

jayam04 committed:
Updates full_stack_tests.yml

jayam04 committed:
Updates full_stack_tests.yml

jayam04 committed:
Added screenshots

jayam04 committed:
Addresses one more comment

jayam04 committed:
Potential Fix: Flakes

jayam04 committed:
Update: full_stack_tests.yml to run each test 10 times in mobile and desktop totalling 20 times

jayam04 committed:
Revert "Addresses one more comment"

This reverts commit 30c924cb74873b3ff7d1a7d17501f722fcd65acf.

jayam04 committed:
Addresses comment

jayam04 committed:
Potential Fix: addBasicAlgebraQuestionToSkill

jayam04 committed:
Fixes some of the potential flakes

imchristie commented on core/tests/puppeteer-acceptance-tests/utilities/user/contributor.ts:
```diff
@@ -336,6 +335,38 @@ export class Contributor extends ExplorationEditor {
+  /**
+   * Selects the badge type in the contribution dashboard.
+   * @param badgeType - The badge type to select.
+   * @param {string} topicName - The name of the topic to select the badge type.
```
This parameter doesn't exist.

jayam04 replied:
Removed

imchristie commented on core/tests/puppeteer-acceptance-tests/utilities/user/contributor.ts:
```diff
@@ -472,6 +507,9 @@ export class Contributor extends ExplorationEditor {
       const rowCells = await row.$$(cellSelector);
       if (rowValues.length !== rowCells.length) {
+        showMessage(
+          `[debug] row values length: ${rowValues.length}, row cells length: ${rowCells.length}`
```
Make sure to remove all these [debug] console log before merging

jayam04 replied:
Done, for this part. Will check for others. Please, don't resolve.

jayam04 replied:
Checked the whole PR. You can resolve it now.

imchristie commented on core/tests/puppeteer-acceptance-tests/utilities/user/exploration-editor.ts:
```diff
@@ -3509,8 +3507,10 @@ export class ExplorationEditor extends BaseUser {
     }
-    await this.clickOn(commitMessage);
-    await this.type(commitMessage, 'Testing Testing');
+    if (commitMessage) {
```
If nothing is pass to this function for the parameter `commitMessage`, it will be "Testing Testing". So it'll always have something right? So this if condition is useless.

jayam04 replied:
We can skip it by using empty sting. Added a comment above.

imchristie commented on core/tests/puppeteer-acceptance-tests/utilities/user/exploration-editor.ts:
```diff
@@ -6172,16 +6178,16 @@ export class ExplorationEditor extends BaseUser {
   }
 
   async fillDescriptionInSaveDraftModal(description: string): Promise<void> {
```
Missing function description

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/utilities/user/practice-question-submitter.ts:
```diff
@@ -634,20 +644,22 @@ export class PracticeQuestionSubmitter extends Contributor {
 
     await this.page.waitForFunction(
       (element: HTMLInputElement | HTMLTextAreaElement, value: string) => {
+        console.log('[debug] element.value: ' + element.value);
```
Make sure to remove all these [debug] console log before merging

jayam04 replied:
Removed.

imchristie reviewed (changes requested):
Completed full review

jayam04 committed:
Fix: Potential flake

jayam04 committed:
Renamed full_stack_tests.yml to test.yml

jayam04 committed:
Revert "removed other tests"

This reverts commit 9e2f527f81aa9737c8063519fca5ef0d7ab35a89.

jayam04 committed:
Adds full_stack_tests.yml

jayam04 committed:
Revert "Updated full_stack_tests.yml."

This reverts commit 454f19305f427c9b52b885b1710dac03a496986e.

jayam04 committed:
Reverts: full_stack_tests.yml

jayam04 committed:
Merge remote-tracking branch 'upstream/develop' into m2.3

jayam04 committed:
Addresses the comment

jayam04 committed:
Lint & Type Fixes

jayam04 committed:
Type and Lint Fixes (2)

jayam04 reviewed (commented):
Addressed comments.

imchristie reviewed (approved):
LGTM

jayam04 committed:
Updates tests by removing known bug

jayam04 committed:
Potential Fix

jayam04 committed:
Updates full_stack_tests.yml

jayam04 committed:
Minor Fixes

jayam04 committed:
Fix: Flakes

jayam04 committed:
Merge branch 'develop' into m2.3

jayam04 committed:
Changes from E2E migration

jayam04 committed:
Fix

jayam04 committed:
Fix

jayam04 committed:
Fix: [Bug] where answer was added to wrong modal and wrong answer was marked as correct.

jayam04 commented:
@imchristie It's regarding last commit [dc821f6](https://github.com/oppia/oppia/pull/23124/commits/dc821f6367dc2112f283e653e163b2bfa7e57a09)

I have changed files, because there was a bug before where the answer in rule editor wasn't added, thus wrong answers were marked as correct. 

Based on the image and screen recording (0:14) we can see that -99 is inserted at different text input than expected.


https://github.com/user-attachments/assets/e42668cb-5e78-4551-994a-f63f989144a2

<img width="978" height="650" alt="image" src="https://github.com/user-attachments/assets/0635631f-becf-4bda-96d2-37a33aaa6211" />

jayam04 committed:
Lint Fixes

jayam04 committed:
Fixes

jayam04 committed:
Removes test.yml

imchristie reviewed (approved):
LGTM.

jayam04 committed:
Fix: Adds more rights to SuperAdminInstance

kevintab95 reviewed (approved):
LGTM for 2 codeowner files. Thanks @jayam04!

masterboy376 reviewed (approved):
@jayam04 LGTM

jayam04 committed:
Merge remote-tracking branch 'upstream/develop' into m2.3

jayam04 committed:
Update create-subtopic-study-guide-with-multiple-sections.spec.ts

jayam04 committed:
Fix interaction tab selection in exploration editor

The commit removes redundant tab switching code since the functionality
was already handled by changeTabInInteractionSelectionModal method.

jayam04 committed:
Fix: Lint errors

jayam04 committed:
Increase timeout for subtopic test to 500000ms

jayam04 commented:
Filed flakes: https://github.com/oppia/oppia/issues/23138#issuecomment-3333665853
https://github.com/oppia/oppia/issues/23158#issuecomment-3333662814

jayam04 committed:
Merge remote-tracking branch 'upstream/develop' into m2.3
